sql: set SearchPath during SET ROLE correctly

otan opened the pull request:
We previously did not set SearchPath on SET ROLE. This is now fixed.

Release justification: bug fix for new feature

Release note: None

rafiss reviewed (approved):
nice find :open_mouth: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69397#-:-Mi-rfdP0JIwI7LIFPGt:bcgpkcp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/user.go, line 471 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69397#-Mi-q67D5qV4C0cKiVkM:-Mi-q67D5qV4C0cKiVkN:bdwh1ju) ([raw file](https://github.com/cockroachdb/cockroach/blob/ed55dabff15714fb7a981fa06c0fdf387bc62201/pkg/sql/user.go#L471)):*
> ```Go
> 	m.paramStatusUpdater.BufferParamStatusUpdate("is_superuser", updateStr)
> 
> 	// The "none" user does resets the SessionUserProto in a SET ROLE.
> ```

nit: typo on "does resets"


<!-- Sent from Reviewable.io -->

otan committed:
sql: set SearchPath during SET ROLE correctly

We previously did not set SearchPath on SET ROLE. This is now fixed.

Release justification: bug fix for new feature

Release note: None
